proposals/date_replace.json

```diff
@@ -0,0 +1,108 @@
+{
+    "id": "date_replace",
```

Comment:
Given `date_get_component`, I think it would be better for symmetry to call this one `date_replace_component`

Reply:
> Given `date_get_component`, I think it would be better for symmetry to call this one `date_replace_component`
I agree